integrations/git_helper_for_declarative_test.go

```diff
@@ -139,6 +139,10 @@ func doGitInitTestRepository(dstPath string) func(*testing.T) {
 	return func(t *testing.T) {
 		// Init repository in dstPath
 		assert.NoError(t, git.InitRepository(dstPath, false))
+		repo, err := git.OpenRepository(dstPath)
+		assert.NoError(t, err)
+		assert.NoError(t, repo.SetDefaultBranch("master"))
+		assert.NoError(t, repo.Close())
```

Comment:
Honestly I wouldn't bother opening the repo here. 

It's a test and we don't need to test SetDefaultBranch - so I'd say just do:

```suggestion
		// forcibly set default branch to master
		_, err := git.NewCommand("symbolic-ref", "HEAD", git.BranchPrefix+"master").RunInDir(dstPath)
		assert.NoError(t, err)
```

This will be quicker and less problem prone than doing anything else.

Reply:
While it seems to have been tried in the past (and reverted from what I understood), wouldn't it cause problems if gitea switch to a non git executable backend in the future ? Using SetDefaultBranch would wrap any change happening there.

Reply:
We can deal with that if it comes to it.

Use symbolic-ref.